specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json

```diff
@@ -0,0 +1,1897 @@
+       },
+       "consumes": [],
+       "produces": [
+        "application/json"
+       ],
+       "parameters": [
+        {
+         "name": "certificateName",
+         "in": "path",
+         "required": true,
+         "type": "string",
+         "description": "Name of the certificate to retrieve."
+        },
+        {
+         "$ref": "#/parameters/subscriptionId"
+        },
+        {
+         "name": "resourceGroupName",
+         "in": "path",
+         "required": true,
+         "type": "string",
+         "description": "Resource group identifier."
+        },
+        {
+         "name": "provisioningServiceName",
+         "in": "path",
+         "required": true,
+         "type": "string",
+         "description": "Name of the provisioning service the certificate is associated with."
+        },
+        {
+         "name": "If-Match",
+         "in": "header",
+         "required": false,
+         "type": "string",
+         "description": "ETag of the certificate."
+        },
+        {
+         "$ref": "#/parameters/api-version"
+        }
+       ],
+       "responses": {
+        "200": {
+         "description": "Metadata for the specified certificate.",
+         "schema": {
+          "$ref": "#/definitions/CertificateResponse"
+         }
+        },
+        "default": {
+         "description": "Default error response.",
+         "schema": {
+          "$ref": "#/definitions/ErrorDetails"
+         }
+        }
+       }
+      },
+      "put": {
+       "tags": [
+        "PUT"
+       ],
+       "summary": "Upload the certificate to the provisioning service.",
+       "description": "Add new certificate or update an existing certificate.",
+       "operationId": "DpsCertificate_CreateOrUpdate",
+       "x-ms-examples": {
+           "DPSCreateOrUpdateCertificate": { "$ref": "./examples/DPSCertificateCreateOrUpdate.json" }
+       },
+       "parameters": [
+        {
+         "$ref": "#/parameters/api-version"
+        },
+        {
+         "$ref": "#/parameters/subscriptionId"
+        },
+        {
+         "name": "resourceGroupName",
+         "in": "path",
+         "description": "Resource group identifier.",
+         "required": true,
+         "type": "string"
+        },
+        {
+         "name": "provisioningServiceName",
+         "in": "path",
+         "description": "The name of the provisioning service.",
+         "required": true,
+         "type": "string"
+        },
+        {
+         "name": "certificateName",
+         "in": "path",
+         "description": "The name of the certificate create or update.",
+         "required": true,
+         "type": "string",
+         "maxLength": 256
+        },
+        {
+         "name": "certificateDescription",
+         "in": "body",
+         "description": "The certificate body.",
+         "required": true,
+         "schema": {
+          "$ref": "#/definitions/CertificateBodyDescription"
```

Comment:
A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable.

Reply:
the request and response are the same but the CertificateResponse includes some system generated properties that are added to CertificateBodyDescription which is why it has this format

Reply:
the PUT is a subset of the GET